### ISSUE TITLE
Web - Search Page - Missing Heading

### ISSUE THREAD

**Issue by jfc3-dol** [USDepartmentofLabor/Comply-Chain#67]
There are no headings on this page. I would expect at least one heading level one (h1) of "Search". 

Search - https://dolcontentdev.opadev.dol.gov/ilab/complychain/build/search

**Comment by KyleLivingstonDOL** [comment]
Hi John, there is no headings because it is a simple search page. Once a user enters in a key word, then a heading pops up with the number of search results. Is not having a heading on a page a 508 compliance issue? Thanks!
Kyle

**Comment by jfc3-dol** [comment]
Screen reader users use headings to job/skip to different sections on the web page. So by having one that says "Search" or "Search Results" if the page also lists the results found they can use the "H" key to go from heading to heading as you would glancing visually at the page.

**Comment by jfc3-dol** [comment]
When the user first goes to the "Search" page, there are no headings. There should be at least one heading level one (h1) telling the user they are on the "Search Results" page.

Once the user searches, there is a heading level one (h1) that givens the number of results and the term searched for, so it's better at least once the user searches.

**Comment by KyleLivingstonDOL** [comment]
@jfc3-dol Thanks John. So to clarify, for 508 compliance, we need to code the SEARCH RESULTS title at the top of the page as H1 and then each heading for the search titles that come up can be coded as H2?

**Comment by jfc3-dol** [comment]
@KyleLivingstonDOL,

Yes, you are correct. Doing this gives the assistive technology user (screen reader) an idea of the pages hierarchy and also allows them to skip from one section to the next as a sighted person would do visually by looking for the next set of bold text.

**Comment by Abychalil** [comment]
Fixed. Search results page heading levels updated to H1 and H2.
![issue 325, 67 search page heading h1,h2](https://user-images.githubusercontent.com/49654920/72354653-1e626700-36b4-11ea-8d8a-7137408f4d99.PNG)

**Comment by Abychalil** [comment]
Addressed this issue. Please retest and verify whether it meets the required 508 standards. 
@jfc3-dol

**Comment by jfc3-dol** [comment]
@Abychalil,
The text "Search" is now a heading level one (h1) as expected, along with the results page names being a heading level two (h2) now too.

I'm wondering why there is a second heading level one (h1) for the text **10 search results for "text"**?

I would expect that to be a heading level two (h2) and then the results headings to be a heading level three (h3).

Please see the attached screenshot.

![comply chain - web - search page - incorrect heading level](https://user-images.githubusercontent.com/10423487/75277224-9d7bac80-57d5-11ea-9e9e-c31348601b87.png)

**Comment by jfc3-dol** [comment]
@Abychalil 
The same heading structure as mentioned above is still there that has two heading level one (h1)items.

**Comment by jfc3-dol** [comment]
@Abychalil @KyleLivingstonDOL 
The same heading structure, as mentioned above (Feb 25th), is still there. There are two heading level one (h1)items when the second one should be a heading level two (h2), and the results page name should be heading level three (h3) items.

**Comment by jfc3-dol** [comment]
@Abychalil 

Why are there what look to be hidden headings in this section of page and duplicates of different headings with two different levels?

Please see the attached screenshot.

![ilab - search page - why are there duplciate headigns with different levels](https://user-images.githubusercontent.com/10423487/88212526-66df7c00-cc25-11ea-92f8-adc06c1dced6.png)


After further looking, it seems that there is one heading level wrapped around another heading. Not sure why there is a heading level one (h1) wrapped around a heading level two (h2)? The same applies to the search result headings where there is a heading level two (h2) wrapped around a heading level three( h3).

I would expect the ones on the outside to be removed, so the heading structure is correct for the page.

Please see the attached screenshot.

![ilab - search page - why are there  headings wrpped around headings](https://user-images.githubusercontent.com/10423487/88212543-6c3cc680-cc25-11ea-9203-96d8f1ae215e.png)

**Comment by jfc3-dol** [comment]
@harpreetsdev @Abychalil @pbhatt17

The same issue mentioned above is still happening. Not sure if the code got sent to us or not. Please check, and I will check with the OPA Tech team to see if they can republish the code to the server, but will wait to hear from you first.

**Comment by harpreetsdev** [comment]
@jfc3-dol 

I don't know why you still seeing this issue - but on my browser the order of Headings from "Search", "Number of Search Results Label" and the "First search result" is following the expected pattern of - H1, H2, H3. 

Attached screenshots below -

![H1SS](https://user-images.githubusercontent.com/13814932/91462396-20022900-e858-11ea-8926-efad834acf57.png)

![H2SS](https://user-images.githubusercontent.com/13814932/91462417-242e4680-e858-11ea-911b-7922ce1e3a32.png)

![H3SS](https://user-images.githubusercontent.com/13814932/91462432-285a6400-e858-11ea-9147-310fe08d9cec.png)


I hope we are on the same page as far as the expectation for this issue. In my opinion the heading order should descend down 
in the order of H1, H2 and H3.

@pbhatt17 @Abychalil

**Comment by jfc3-dol** [comment]
@harpreetsdev @Abychalil @pbhatt17

Maybe it's because it wasn't pushed with the last set of code yet. I can check again next week after we get the latest Sprint code in the next few days.

**Comment by harpreetsdev** [comment]
@jfc3-dol 

Sounds good. No Rush.

@pbhatt17 @Abychalil

**Comment by harpreetsdev** [comment]
@jfc3-dol 

Please test this as well

@pbhatt17 @Abychalil

**Comment by jfc3-dol** [comment]
@harpreetsdev @pbhatt17 @Abychalil

Please see the attached screenshots that show one heading level wrapped around the second one. I would expect the heading level one (h1) that goes around the heading level two (h2) to be removed and the heading level two (h2) that goes around the heading level three (h3) to be removed for the search results. 

So once fixed, I would expect one heading level two (h2) around the example text **10 search results for "test"** and then only heading level three (h3) around each set of results.


![ilab - comply chain - search page - heading level one  wrapped around a heading level two](https://user-images.githubusercontent.com/10423487/92161271-629aa700-edfe-11ea-966e-a4a40957d208.png)


![ilab - comply chain - search page - heading level twowrapped around a heading level three](https://user-images.githubusercontent.com/10423487/92161269-62021080-edfe-11ea-80ec-dfd34c624bcf.png)

**Comment by harpreetsdev** [comment]
@jfc3-dol 
What you are seeing are not duplicate or second h1, h2 tags, these are closing HTML tags for the same tags opened above. If you see closely the opening tags are <h1> and the closing tags are </h1> with a forward slash. But they represent 1 tag only. These are not second tags.
Also the wrapping you see around the h2 tags is how the HTML is rendering the text.

@pbhatt17 @Abychalil

**Comment by jfc3-dol** [comment]
@harpreetsdev @pbhatt17 @Abychalil

There should not be a heading level two (h2) around a heading level three (h3) is what I'm saying. Text should not have more than one heading level, which is the problem.

Each heading should only have one heading level, not two, which is what I'm pointing out.

**Comment by harpreetsdev** [comment]
@jfc3-dol 
Maybe there is an issue with how the tags are being rendered. Let us look at it again and get back to you.

@pbhatt17 @Abychalil

**Comment by jfc3-dol** [comment]
@harpreetsdev @pbhatt17 @Abychalil

Thanks, please check both the issues pointed out above.

**Comment by jfc3-dol** [comment]
@harpreetsdev @pbhatt17 @Abychalil

There is only one heading around each set of heading text now. I even turned on VoiceOver for iOS to check that it was read correctly there too, and it was.

So I'm going to close this issue.

Please see the attached screenshot.

![ilab - comply chain - search page - There is only one heading around each set of heading text now](https://user-images.githubusercontent.com/10423487/107063356-4681a400-67a8-11eb-8b12-ff1eb5c4353b.png)
